### PR TITLE
feat(helm): Add Helm module

### LINKED ISSUE #1443: Helm Support
#### Describe the solution you'd like
Helm as simply as possible is a package manager for Kubernetes. We simply deploy applications to the Kubernetes using Helm's package format called Chart.

Sometimes we might need to know about Helm's version that I am using on Kubernetes cluster. We can simply get the version information of Helm with version command like below:

```
$ helm version
version.BuildInfo{Version:"v3.2.1", GitCommit:"fe51cd1e31e6a202cba7dead9552a6d418ded79a", GitTreeState:"clean", GoVersion:"go1.13.10"}
```

I thought it might be good If I could see the Helm version on my terminal that I am using with the Helm's logo.

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Furkan <furkan.turkal@hotmail.com>

#### Motivation and Context

Closes #1443 

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

**Comment by Dentrax** [comment]
Hey @andytom, we have tried to add Helm support as we talked about at #1443. But somehow, these tests does just fail by Ubuntu. Even so, tests are passing by `nightly` [unlike](https://github.com/starship/starship/runs/856881272?check_suite_focus=true) `stable`, that's where we couldn't figure it out why. Thanks! :)

**Comment by andytom** [comment]
> Hey @andytom, we have tried to add Helm support as we talked about at #1443. But somehow, these tests does just fail by Ubuntu. Even so, tests are passing by `nightly` [unlike](https://github.com/starship/starship/runs/856881272?check_suite_focus=true) `stable`, that's where we couldn't figure it out why. Thanks! :)

Unfortunately some of our tests are a bit unstable, see #886, we are trying to figure out the root cause but not quite there yet. I'll rerun the tests as that normally fixes the issue.

File: src/modules/helm.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	// v3.1.1+gafe7058
+
+	let version = helm_stdout
+		// split into ["v3.1.1","gafe7058"]
+		.splitn(2, '+')
+		// return "v3.1.1"
+		.next()?;
+
+	Some(version.to_string())
+}
```
Comment (andytom) [inline src/modules/helm.rs:69]: Can you update this function to also support helm 2, this should just be a case of removing the prefixed `Client: ` from the version string.
```
helm version --short --client
Client: v2.16.9+g8ad7037
```

File: src/modules/helm.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		let expected = Some(format!("via {} ", Color::Blue.bold().paint("⎈ v3.1.1")));
+		assert_eq!(expected, actual);
+		dir.close()
+	}
+
+	#[test]
+	fn test_format_helm_version() {
+		let input = "v3.1.1+ggit afe7058";
+		assert_eq!(format_helm_version(input), Some("v3.1.1".to_string()));
+	}
```
Comment (andytom) [inline src/modules/helm.rs:118]: Maybe change this to also test for helm2 versions.

 ```suggestion
	fn test_format_helm_version() {
		assert_eq!(format_helm_version("v3.1.1+gafe7058"), Some("v3.1.1".to_string()));
		assert_eq!(format_helm_version("Client: v2.16.9+g8ad7037"), Some("v2.16.9".to_string()));
	}
```

File: src/modules/helm.rs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+/// Creates a module with the current Helm version
+///
+/// Will display the Helm version if any of the following criteria are met:
+///     - Current directory contains a `helmfile.yaml` file
+///     - Current directory contains a `Chart.yaml` file
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+	let is_helm_project = context
+		.try_begin_scan()?
+		.set_files(&["helmfile.yaml", "Chart.yaml"])
+		.set_folders(&["templates"])
```
[the remark this answers is not in the document]
Comment (Dentrax) [reply src/modules/helm.rs:16]: In this case, i don't have to check any folders anymore, right?

File: src/modules/helm.rs (hunk 1 above)
Comment (Dentrax) [reply src/modules/helm.rs]: I just changed this parser logic, now should be support both Helm 1 and 2 version commands, i guess. :)

File: src/modules/helm.rs (hunk 2 above)
Comment (Dentrax) [reply src/modules/helm.rs]: Done

File: docs/config/README.md [hunk 4]
```diff
@@ -1150,6 +1151,79 @@ trim_at = ".companyname.com"
...
+
+- The current directory contains a `helmfile.yaml` file
+- The current directory contains a `Chart.yaml` file
+
+### Options
+
+| Option     | Default                            | Description                                    |
+| ---------- | ---------------------------------- | ---------------------------------------------- |
+| `format`   | `"via [$symbol$version]($style) "` | The format for the module.                     |
+| `symbol`   | `"🐹 "`                            | A format string representing the symbol of Helm. |
```
Comment (cappyzawa) [inline docs/config/README.md:1167]: based on https://github.com/starship/starship/pull/1463/files#diff-3009cc0091e546b48a6d01d63548e923R17.

```suggestion
| `symbol`   | `"⎈ "`                            | A format string representing the symbol of Helm. |
```

Comment (Dentrax) [reply docs/config/README.md]: Updated

File: src/modules/helm.rs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        let expected = Some(format!("via {} ", Color::Blue.bold().paint("⎈ v3.1.1")));
+        assert_eq!(expected, actual);
+        dir.close()
+    }
+
+    #[test]
+    fn test_format_helm_version() {
+        let helm_1 = "v3.1.1+ggit afe7058";
+        let helm_2 = "Client: v2.16.9+g8ad7037";
+        assert_eq!(format_helm_version(helm_1), Some("v3.1.1".to_string()));
```
Comment (andytom) [inline src/modules/helm.rs:124]: Minor consistency correction here.

```suggestion
        let helm_3 = "v3.1.1+ggit afe7058";
        let helm_2 = "Client: v2.16.9+g8ad7037";
        assert_eq!(format_helm_version(helm_3), Some("v3.1.1".to_string()));
```

File: docs/config/README.md [hunk 6]
```diff
@@ -1150,6 +1151,79 @@ trim_at = ".companyname.com"
...
+### Example
+
+```toml
+# ~/.config/starship.toml
+
+[helm]
+format = "via [⎈ $version](bold blue) "
+```
+
+## Hostname
```
Comment (andytom) [inline docs/config/README.md:1190]: It looks like you have duplicated the hostname entry here.

File: src/modules/helm.rs (hunk 5 above)
Comment (Dentrax) [reply src/modules/helm.rs]: Done

File: docs/config/README.md (hunk 6 above)
Comment (Dentrax) [reply docs/config/README.md:1154]: 👍 some copy-paste problems. :) removed.

**Comment by Dentrax** [comment]
> Have added a few more suggestions, something else to think about is the default symbol and style are very similar to the defaults for the kubernetes module which I think a lot of helm users will also be using, can you think of any other style and symbols we could use?

I think you are right. Both of looks similar. But I couldn't find any other nice symbol unicodes. :(

**Comment by andytom** [comment]
> > Have added a few more suggestions, something else to think about is the default symbol and style are very similar to the defaults for the kubernetes module which I think a lot of helm users will also be using, can you think of any other style and symbols we could use?
> 
> I think you are right. Both of looks similar. But I couldn't find any other nice symbol unicodes. :(

How about changing the default color to white maybe that will be enough?

**Comment by Dentrax** [comment]
> I've been testing this locally and just realised there are a few other little changes that need to be made. You will need to add the module to the `PROMPT_ORDER` in `src/configs/starship_root.rs` and `ALL_MODULES` in `src/module.rs`.

Fixed. :)



> > > Have added a few more suggestions, something else to think about is the default symbol and style are very similar to the defaults for the kubernetes module which I think a lot of helm users will also be using, can you think of any other style and symbols we could use?
> > 
> > 
> > I think you are right. Both of looks similar. But I couldn't find any other nice symbol unicodes. :(
> 
> How about changing the default color to white maybe that will be enough?

Yep, White seems to be more appropriate for this case.

File: docs/config/README.md [hunk 7]
```diff
@@ -1114,6 +1115,42 @@ The module will be shown if any of the following conditions are met:
...
+
+\*: This variable can only be used as a part of a style string
+
+### Example
+
+```toml
+# ~/.config/starship.toml
+
+[helm]
+format = "via [⎈ $version](bold blue) "
```
Comment (cappyzawa) [inline docs/config/README.md:1151]: Based on https://github.com/starship/starship/pull/1463/files#diff-3009cc0091e546b48a6d01d63548e923R18

```suggestion
format = "via [⎈ $version](bold white) "
```

**Comment by andytom** [comment]
Thanks for your contribution @Dentrax and @developer-guy and thank you for reviewing @cappyzawa.

**Pull request merged**